sunpy/net/helioviewer.py

```diff
@@ -127,29 +168,36 @@ def download_jp2(self, date, directory=None, overwrite=False, **kwargs):
         >>> import sunpy.map
         >>> from sunpy.net import helioviewer
         >>> hv = helioviewer.HelioviewerClient()  # doctest: +REMOTE_DATA
-        >>> filepath = hv.download_jp2('2012/07/03 14:30:00', observatory='SDO', instrument='AIA', detector='AIA', measurement='171')   # doctest: +REMOTE_DATA
+        >>> filepath = hv.download_jp2('2012/07/03 14:30:00', observatory='SDO', instrument='HMI', detector='None', measurement='continuum')   # doctest: +REMOTE_DATA
         >>> aia = sunpy.map.Map(filepath)   # doctest: +REMOTE_DATA
         >>> aia.peek()   # doctest: +SKIP
-
-        >>> data_sources = hv.get_data_sources()  # doctest: +REMOTE_DATA
-        >>> file = hv.download_jp2('2012/07/03 14:30:00', sourceId=data_sources['SOHO']['LASCO']['C2']['white-light']['sourceId'])   # doctest: +REMOTE_DATA
         """
+        global datasource_info
+
+        if('observatory' in kwargs):
+            observatory = kwargs['observatory'] if(kwargs['observatory']) else 'None' 
+            instrument = kwargs['instrument'] if(kwargs['instrument']) else 'None'
+            detector = kwargs['detector'] if(kwargs['detector']) else 'None'
+            measurement = kwargs['measurement'] if(kwargs['measurement']) else 'None'
+            key = (observatory, instrument, detector, measurement)
+            sourceid = datasource_info[key]
+
         params = {
             "action": "getJP2Image",
-            "date": self._format_date(date)
+            "date": self._format_date(date),
+            "sourceId" : sourceid
         }
         params.update(kwargs)
-
         # JPIP URL response
         if 'jpip' in kwargs:
             return self._get_json(params)
 
         return self._get_file(params, directory, overwrite=overwrite)
 
-    def download_png(self, date, image_scale, layers, directory=None,
-                     overwrite=False, **kwargs):
-        """Downloads a PNG image using data from Helioviewer.org.
-
+    def download_png(self, date, image_scale, layers, eventLabels=False, events=None, 
+                     scale=False, watermark=False, directory=None, overwrite=False, **kwargs):
+        """Downloads a PNG image using data from Helioviewer.org. It uses the
```

Comment:
If I recall, multi line strings should have a new line break between the `"` and the text.